src/test/java/reactor/core/publisher/MonoSequenceEqualTest.java

```diff
@@ -15,11 +15,211 @@
  */
 package reactor.core.publisher;
 
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.junit.Assert;
 import org.junit.Test;
+import org.reactivestreams.Subscription;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.StepVerifier;
+import reactor.util.concurrent.QueueSupplier;
 
 public class MonoSequenceEqualTest {
 
 	@Test
-	public void normal() {
+	public void sequenceEquals() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three"),
+						Flux.just("one", "two", "three")))
+		            .expectNext(Boolean.TRUE)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void sequenceLongerLeft() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three", "four"),
+						Flux.just("one", "two", "three")))
+		            .expectNext(Boolean.FALSE)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void sequenceLongerRight() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three"),
+						Flux.just("one", "two", "three", "four")))
+		            .expectNext(Boolean.FALSE)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void sequenceErrorsLeft() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two").concatWith(Mono.error(new IllegalStateException())),
+						Flux.just("one", "two", "three")))
+		            .verifyError(IllegalStateException.class);
+	}
+
+	@Test
+	public void sequenceErrorsRight() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three"),
+						Flux.just("one", "two").concatWith(Mono.error(new IllegalStateException()))))
+		            .verifyError(IllegalStateException.class);
+	}
+
+	@Test
+	public void sequenceErrorsBothPropagatesLeftError() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three").concatWith(Mono.error(new IllegalArgumentException("left"))).hide(),
+						Flux.just("one", "two").concatWith(Mono.error(new IllegalArgumentException("right"))).hide()))
+		            .verifyErrorMessage("left");
```

Comment:
the two sources are drained until termination so that the error can be detected and propagated. thus it's just a matter that the first inner subscriber being in an error state is inspected first, and its error is the one that gets propagated (despite the shorter "right" source)